modules/channel-messenger/src/backend/messenger.ts

```diff
@@ -173,33 +173,20 @@ export class MessengerService {
         await bot.client.sendAction(senderId, 'mark_seen')
 
         if (webhookEvent.message) {
-          await this._sendEvent(bot.botId, senderId, pageId, webhookEvent.message, { type: 'message' })
+          await this._sendEvent(bot.botId, senderId, webhookEvent.message.text)
         } else if (webhookEvent.postback) {
-          await this._sendEvent(
-            bot.botId,
-            senderId,
-            pageId,
-            { text: webhookEvent.postback.payload },
-            { type: 'callback' }
-          )
+          await this._sendEvent(bot.botId, senderId, webhookEvent.postback.payload.text)
         }
       }
     }
   }
 
-  private async _sendEvent(botId: string, senderId: string, pageId: string, message: any, args: { type: string }) {
-    await this.bp.events.sendEvent(
-      this.bp.IO.Event({
-        botId,
-        channel: 'messenger',
-        direction: 'incoming',
-        payload: message,
-        preview: message.text,
-        threadId: pageId,
-        target: senderId,
-        ...args
-      })
-    )
+  private async _sendEvent(botId: string, senderId: string, text: string) {
+    const conversation = await this.bp.experimental.conversations.forBot(botId).recent(senderId)
+
+    await this.bp.experimental.messages
+      .forBot(botId)
+      .receive(conversation.id, { type: 'text', text }, { channel: 'messenger' })
```

Comment:
Small change here from previous behaviour : we create a text payload instead of taking the entire message payload from messenger